apps/scratch/src/components/TaskConfig.tsx

```diff
@@ -26,33 +29,57 @@ const TaskConfig = ({
         return;
       }
 
-      config.canEditTaskBlocks = canEditTaskBlocks;
+      updateFunction(config);
+
+      // update toolbox
+      window.dispatchEvent(new UpdateBlockToolboxEvent());
+    },
+    [vm],
+  );
+
+  const onAllowAllBlocks = useCallback(
+    (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
+      updateConfig(e, (config) => (config.allowedBlocks = allowAllBlocks));
+    },
+    [updateConfig],
+  );
+
+  const onAllowNoBlocks = useCallback(
+    (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) =>
+      updateConfig(e, (config) => (config.allowedBlocks = allowNoBlocks)),
+    [updateConfig],
+  );
+
+  const onSubmit = useCallback(
+    (e: FormEvent<HTMLFormElement>) => {
+      e.preventDefault();
+      e.stopPropagation();
+
+      updateConfig(e, () => {});
 
       hideModal();
     },
-    [vm, canEditTaskBlocks],
+    [vm],
   );
 
   useEffect(() => {
-    // every time the modal is opened, load the latest config values
-    setCanEditTaskBlocks(vm.crtConfig?.canEditTaskBlocks ?? true);
+    // every time the modal is opened, load the form values based on the config
   }, [isShown]);
 
   return (
     <Modal isShown={isShown}>
       <h1>Task Config</h1>
 
       <form onSubmit={onSubmit} data-testid="task-config-form">
-        <label>
-          <span>Can the blocks provided with the task be edited?</span>
-          <input
-            type="checkbox"
-            min="0"
-            checked={canEditTaskBlocks}
-            onChange={(e) => setCanEditTaskBlocks(e.target.checked)}
-            data-testid="can-edit-task-blocks-checkbox"
-          />
-        </label>
+        <button
+          onClick={onAllowAllBlocks}
+          data-testid="allow-all-blocks-button"
+        >
+          Allow all blocks to be used
+        </button>
+        <button onClick={onAllowNoBlocks} data-testid="allow-no-blocks-button">
+          Disallow any block to be used
```

Comment:
I fear localizing stuff in the scratch app will be a gigantic pain.. I am trying to ignore that for now but it would be good to look into that, although that should most definitely be its own PR. Since scratch already does localize stuff, we need to some some funky stuff to merge our translations with theirs but it should be doable with some magic :)